core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -3224,4 +3224,25 @@ private SqlSpecialOperatorWithPolicy(String name, SqlKind kind, int prec, boolea
   @Test void testSimplifyVarbinary() {
     checkSimplifyUnchanged(cast(cast(vInt(), tVarchar(true, 100)), tVarbinary(true)));
   }
+
+  @Test void testSimplifySimpleArithmetic() {
+    RexNode a = vIntNotNull(1);
+    RexNode zero = literal(0);
+    RexNode one = literal(1);
+
+    RexNode b = vDecimalNotNull(2);
+    RexNode half = literal(new BigDecimal(0.5), b.getType());
+
+    checkSimplify(add(a, zero), "?0.notNullInt1");
```

Comment:
Could you please add a test like `add(zero, sub(vInt(0), vInt(0)))`?
The expression should evaluate to null in case `vInt(0)` is null.

Reply:
Good point. Thanks. @vlsi 
I have added some test cases concerning null. Please check.